fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/responses/RdfStreamStreamingOutput.java

```diff
@@ -113,75 +108,84 @@ public RdfStreamStreamingOutput(final RdfStream rdfStream, final Map<String, Str
+            } else {
+                RDFDataMgr.write(output, model.getGraph(), dataFormat);
+            }
+        }
     }
 
-    protected static final Function<? super Triple, Statement> toStatement = t -> {
-        final Resource subject = getResourceForSubject(t.getSubject());
-        final URI predicate = vfactory.createURI(t.getPredicate().getURI());
-        final Value object = getValueForObject(t.getObject());
-        return vfactory.createStatement(subject, predicate, object);
-    };
-
-    private static Resource getResourceForSubject(final Node subjectNode) {
-        return subjectNode.isBlank() ? vfactory.createBNode(subjectNode.getBlankNodeLabel())
-                : vfactory.createURI(subjectNode.getURI());
-    }
 
-    protected static Value getValueForObject(final Node object) {
-        if (object.isURI()) {
-            return vfactory.createURI(object.getURI());
-        } else if (object.isBlank()) {
-            return vfactory.createBNode(object.getBlankNodeLabel());
-        } else if (object.isLiteral()) {
-            final String literalValue = object.getLiteralLexicalForm();
-
-            final String literalDatatypeURI = object.getLiteralDatatypeURI();
-
-            if (!object.getLiteralLanguage().isEmpty()) {
-                return vfactory.createLiteral(literalValue, object.getLiteralLanguage());
-            } else if (literalDatatypeURI != null) {
-                final URI uri = vfactory.createURI(literalDatatypeURI);
-                return vfactory.createLiteral(literalValue, uri);
+    /*
+     * Convert an RdfStream to a JSON object suitable for writing by JsonUtils::write
+     *
+     * Note: we use the jsonld library directly here instead of using Jena's facade
+     * because Jena doesn't permit writing expanded or flattened formats.
+     * Once https://github.com/apache/jena/pull/139 is merged, much of this code will go away
+     */
+    private static Object rdfToJsonLd(final RdfStream rdfStream, final Map<String, String> prefixes,
+            final String profile) throws IOException, JsonLdError {
+        final RDFDataset ds = new RDFDataset();
+        rdfStream.forEach(triple -> {
+            final Node s = triple.getSubject();
+            final Node p = triple.getPredicate();
+            final Node o = triple.getObject();
+            if (o.isLiteral()) {
+                ds.addTriple(s.getURI(), p.getURI(), o.getLiteralLexicalForm(), o.getLiteralDatatypeURI(),
+                    o.getLiteralLanguage());
             } else {
-                return createLiteral(vfactory, object.getLiteralValue());
+                ds.addTriple(s.getURI(), p.getURI(), o.getURI());
             }
+        });
+        final JsonLdOptions opts = new JsonLdOptions();
+        opts.setCompactArrays(true);
+        final Object json = new JsonLdApi(opts).fromRDF(ds);
+        if (profile.equals("http://www.w3.org/ns/json-ld#compacted")) {
```

Comment:
Should these format URIs be constants? Or are there already constants for them in our gigantic sack of dependencies?